src/components/atoms/FormattedInput/FormattedInput.tsx

```diff
@@ -44,6 +45,8 @@ const FormattedInput = forwardRef<HTMLInputElement, FromattedInputProps>(
     ref,
   ) => {
     const [parsedValue, setParsedValue] = useState<NanValue>(value || "");
+    const parsedValueRef = useRef<NanValue>(parsedValue);
+    parsedValueRef.current = parsedValue;
```

Comment:
`parsedValueRef` will not be updated on call of `setParsedValue`. use this hook to keep the ref and state in sync https://github.com/streamich/react-use/blob/master/docs/useLatest.md

Reply:
`import { useRef } from 'react';

const useLatest = <T>(value: T): { readonly current: T } => {
  const ref = useRef(value);
  ref.current = value;
  return ref;
};

export default useLatest;`

This hook seems to do the exact same thing. 
Why won't the ref's value change when component re-renders on state change ?